src/csharp/Microsoft.Spark/Interop/Ipc/JvmBridge.cs

```diff
@@ -241,6 +248,10 @@ private ISocketWrapper GetConnection()
                 socket?.Dispose();
                 throw;
             }
+            finally
+            {
+                _socketSemaphore.Release();
+            }
```

Comment:
I think in rare cases we may still hit this deadlock.  Say if max backend threads is 10.  And if we are currently using 10 connections (0 in the _sockets queue) we hit this exception.  The `socket?.Dispose()` will not requeue this socket and we will release the semaphore.  Then in the next thread (assuming the other 9 connections are still busy), a new connection will be made to the backend and this connection may be scheduled on the blocked netty thread.

Reply:
> I think in rare cases we may still hit this deadlock.

Can we fix it?

Reply:
Or not? If not, can you add the comment to the code?

Reply:
Should we reuse the `socket` if it's still "valid"?

Reply:
I think we should just leave a comment in the code and see if netty 5 will prevent this from happening.

Reply:
Requeuing the socket if we received a JvmException from the DotnetBackendHandler.